presto-main-base/src/main/java/com/facebook/presto/sql/planner/optimizations/GroupInnerJoinsByConnectorRuleSet.java

```diff
@@ -343,21 +336,22 @@ protected PlanNode getCombinedJoin(JoinNode node, FunctionResolution functionRes
             Lookup lookup = context.getLookup();
             Session session = context.getSession();
             PlanNodeIdAllocator idAllocator = context.getIdAllocator();
-            VariableAllocator variableAllocator = context.getVariableAllocator();
+//            VariableAllocator variableAllocator = context.getVariableAllocator();
 
             MultiJoinNode groupInnerJoinsMultiJoinNode = new JoinNodeFlattener(node, functionResolution, determinismEvaluator, lookup).toMultiJoinNode();
             MultiJoinNode rewrittenMultiJoinNode = joinPushdownCombineSources(groupInnerJoinsMultiJoinNode, idAllocator, metadata, session, lookup);
             if (rewrittenMultiJoinNode.getContainsCombinedSources()) {
                 // Create a left deep join tree
-                PlanNode leftDeepJoinTree = createLeftDeepJoinTree(rewrittenMultiJoinNode, idAllocator);
+//                PlanNode leftDeepJoinTree = createLeftDeepJoinTree(rewrittenMultiJoinNode, idAllocator);
                 // Push pulled up predicates to re-form the Join conditions and remove CrossJoins
-                return predicatePushdownOptimizer.optimize(
-                        leftDeepJoinTree,
-                        session,
-                        TypeProvider.viewOf(variableAllocator.getVariables()),
-                        variableAllocator,
-                        idAllocator,
-                        context.getWarningCollector()).getPlanNode();
+//                return predicatePushdownOptimizer.optimize(
```

Comment:
Instead of moving this code out, can you try building a `SimplePlanRewriter` that walks the `leftDeepJoinTree` and replaces all the `GroupReference` nodes with their actual plan nodes by using the `lookup` ? By doing this, we can continue to use the `predicatePushdownOptimizer` here